### PR TITLE
replace actool with nix + python builder

### PR DESCRIPTION
**Pull request by Mic92**
By using nix, the manifest can be directly generated from nix expressions.
 When acbuild was used all dependencies had to be first copied into .acbuild
 before being compressed, the python script however build up a tar command,
 which creates the compressed archive in one shot. When building thin images,
 a shell script had to execute acbuild multiple times to add mounts. The python
 script however allows to update the json manifest without executing additional
 processes.

I was using python in the first place, because I assume it is often already installed and my original plan was to use the `tarfile` module to create an archive. However it turns out that creating tar files with python is way slower than using `tar` directly.

### CODE REVIEW

**Comment by steveej** [comment]
I like the JSON handling using nix a lot, so I definitely want to merge this in some way or shape.
I'm not 100% sure what we gain from the python script, since packing up is done via the external `tar` utility. Could you remind me again of why we need the script?

**Comment by Mic92** [comment]
Parsing/serializing json mainly. 
The result of exportReferencesGraph is only accessible at build time and this information needs to be added to the manifest as well as to the tar archive.

File: lib/mkACI.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    (options, args) = parser.parse_args()
+    if len(args) < 3:
+        sys.stderr.write("not enough arguments")
+        parser.print_help()
+        sys.exit(1)
+    return options, args
+
+
+def paths_from_graphs(graphs):
+    paths = {}
```
Comment (steveej) [inline lib/mkACI.py:25]: since you're using this dict as a set, you might as well use a set ;-)

File: lib/mkACI.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+              manifest,
+              custom_env,
+              closures,
+              thin=True,
+              static=False,
+              dnsquirks=False):
+    os.mkdir(os.path.dirname(aci))
+    nixstore = []
+    envfiles = []
+    if static:
```
Comment (steveej) [inline lib/mkACI.py:86]: the assumption is that there's only one path in _closures_ if the build is static, correct?

File: lib/mkACI.py (hunk 1 above)
Comment (Mic92) [inline lib/mkACI.py:25]: what is the benefit? python uses a dict as well internally and I don't use any set operations such as union - the amount of code would be the same. http://svn.python.org/view/python/trunk/Lib/sets.py?revision=77310&view=markup

File: lib/mkACI.py (hunk 2 above)
Comment (Mic92) [inline lib/mkACI.py:86]: I did the same thing as the old code here: https://github.com/steveeJ/nix2aci/pull/5/files#diff-83899cfd5c3f6fe16d157e4299c8c684L104

File: lib/mkACI.py (hunk 1 above)
Comment (steveej) [inline lib/mkACI.py:25]: Looks like the only benefit would be to maintain the illusion of a true set type, probably a matter of taste in this case.

**Pull request merged**